client/app/main/student/student.controller.js

```diff
@@ -109,6 +109,50 @@ function StudentCtrl($scope, $stateParams, Intervention, Modal, Outreach,
       toastr.error(err);
     });
   };
+  $scope.toggleArchiveOutreach = function(outreach) {
+    var oldValue = outreach.archived;
+    var promise = Outreach.toggleArchive({
+      id: outreach._id
+    }, {
+      archived: !(outreach.archived)
+    }).$promise;
+    promise.then(function(outreach) {
+      $scope.$evalAsync(function() {
+        var outreaches = $scope.student.outreaches
+        outreaches.map(function(oldOutreach, i) {
+            if(oldOutreach._id === outreach._id) {
+              outreaches[i].archived = outreach.archived;
+            }
+        });
+      });
+      toastr.info(
+        'The ' + outreach.type +
+        ' outreach has been' +
+        (outreach.archived === true ? ' archived.' : ' unarchived.'));
+    }, function(err) {
+      outreach.archived = oldValue;
+      toastr.error(err);
+    });
+  };
+  $scope.deleteOutreach = function(outreach) {
+    var model = outreach;
+    var deleteFn = function(model) {
+      Outreach.delete({id: model._id});
+      var outreaches = $scope.student.outreaches
+      outreaches.map(function(oldOutreach, i) {
```

Comment:
Same comment about using map here as above. In this case we probably want to use lodash's pull function.